### PR TITLE
add option for SASL auth

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies

Add ability for SASL auth

## This introduces a breaking change

- [ ] Yes
- [x] No

## This should yield a new module release

- [x] Yes
- [ ] No

## This Pull Request implements

SASL authentication for IBM kafka cluster

### CODE REVIEW

File: thoth/messaging/message_base.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def __init__(
         ssl_auth: int = 1,
```
Comment (harshad16) [inline thoth/messaging/message_base.py:55]: is this attribute required anywhere?

**Review by harshad16 — COMMENTED** [review]
lgtm,
thanks for the addition.

File: thoth/messaging/message_base.py (hunk 1 above)
Comment (KPostOffice) [reply thoth/messaging/message_base.py:55]: No, I removed it

**Review by harshad16 — APPROVED** [review]
lgtm

**Pull request merged**